### ISSUE TITLE
[java] CloseResource possible false positive with Primitive Streams

### ISSUE THREAD

**Issue by BerndFarkaDyna** [pmd/pmd#2402]
**Affects PMD Version:** 6.22.0

**Rule:**

[CloseResource](https://pmd.github.io/latest/pmd_rules_java_errorprone.html#closeresource)

**Description:**

In Issue #1922 only java.util.Stream got handled, now the same false/positives occure on IntStream, LongStream and DoubleStream

**Code Sample demonstrating the issue:**

```
public class PmdPrimitiveStreamReproducer {

    public void reproduceIntStream() {
        IntStream iStream = IntStream.of(1).filter(i -> i < 5);
        iStream.anyMatch(i -> i < 5);
    }


    public void reproduceLongStream() {
        LongStream lStream = LongStream.of(1).filter(i -> i < 5);
        lStream.anyMatch(i -> i < 5);
    }


    public void reproduceDoubleStream() {
        DoubleStream dStream = DoubleStream.of(1).filter(i -> i < 5);
        dStream.anyMatch(i -> i < 5);
    }

}

```

**Expected outcome:** no violation should be thrown

*   Does PMD report a violation, where there shouldn't be one? -> false-positive

**Running PMD through:** *[ Gradle | IntelliJ]*

**Comment by BerndFarkaDyna** [comment]
@adangel : i created https://github.com/pmd/pmd/pull/2403 to fix this problem...
